crates/datafusion/src/lib.rs

```diff
@@ -143,12 +143,36 @@ impl TableProvider for HudiDataSource {
     }
 }
 
+pub struct HudiTableFactory;
+
+#[async_trait]
+impl TableProviderFactory for HudiTableFactory {
+    async fn create(
+        &self,
+        _state: &dyn Session,
```

Comment:
should also take configs from session?

Reply:
I don't see this being used in other implementations (PostgreSQL and duckdb):
https://github.com/datafusion-contrib/datafusion-table-providers/blob/ce93f2c916fcd57d1d5b7339e64963c3a72d3673/src/postgres.rs#L199

https://github.com/datafusion-contrib/datafusion-table-providers/blob/ce93f2c916fcd57d1d5b7339e64963c3a72d3673/src/duckdb.rs#L208

passing it here feels redundant as TableProvider::scan already provides a session, but I'm not sure really, this seem to work as is